src/core/resources/search/tokenSearch.ts

```diff
@@ -90,43 +86,14 @@ async function tokenSearchQueryFunction({
   const url = `/${chainId}/?${qs.stringify(queryParams)}`;
   try {
     const tokenSearch = await tokenSearchHttp.get<{ data: SearchAsset[] }>(url);
-    return parseTokenSearch(tokenSearch.data.data, chainId);
+    return tokenSearch.data.data.map((asset) =>
+      parseTokenSearch(asset, chainId),
+    );
   } catch (e) {
     return [];
   }
 }
 
-function parseTokenSearch(assets: SearchAsset[], chainId: ChainId) {
-  return assets
-    .map((a) => {
-      const networkInfo = a.networks[chainId];
-
-      const asset: SearchAsset = {
-        ...a,
-        address: networkInfo ? networkInfo.address : a.address,
-        chainId,
-        decimals: networkInfo ? networkInfo.decimals : a.decimals,
-        isNativeAsset: [
-          `${ETH_ADDRESS}_${ChainId.mainnet}`,
-          `${ETH_ADDRESS}_${ChainId.optimism}`,
-          `${ETH_ADDRESS}_${ChainId.arbitrum}`,
-          `${BNB_BSC_ADDRESS}_${ChainId.bsc}`,
-          `${MATIC_POLYGON_ADDRESS}_${ChainId.polygon}`,
-          `${ETH_ADDRESS}_${ChainId.base}`,
-          `${ETH_ADDRESS}_${ChainId.zora}`,
-          `${ETH_ADDRESS}_${ChainId.avalanche}`,
-          `${ETH_ADDRESS}_${ChainId.blast}`,
-          `${ETH_ADDRESS}_${ChainId.degen}`,
-        ].includes(`${a.uniqueId}_${chainId}`),
```

Comment:
replaced this with `isNativeAsset` from `~/core/utils/chains`, fixes some chains like degen that was not getting the right native